Source/Core/CesiumTerrainProvider.js

```diff
@@ -246,7 +250,7 @@ define([
         }
 
         function requestMetadata() {
-            var metadata = loadJson(metadataUrl);
+            var metadata = RequestScheduler.request(metadataUrl, loadJson);
```

Comment:
Looking at this now, I suspect this is a better approach than doing this implicitly/explicitly in the utility functions.

It's explicit to the user, should extend well to new request functions, and I imagine is what a typical game engine's "resource manager" looks like.